src/gpu/nvidia/cudnn_convolution.cpp

```diff
@@ -33,47 +34,44 @@ status_t cudnn_convolution_fwd_t::execute_convolution(
         using scratch_acc_t = ::sycl::accessor<uint8_t, 1,
                 ::sycl::access::mode::read_write>;
-        auto x_acc = CTX_IN_ACCESSOR(DNNL_ARG_SRC);
-        auto weights_acc = CTX_IN_ACCESSOR(DNNL_ARG_WEIGHTS);
-        auto y_acc = CTX_OUT_ACCESSOR(DNNL_ARG_DST);
-        std::shared_ptr<
-                ::sycl::accessor<uint8_t, 1, ::sycl::access::mode::read>>
-                bias_acc;
+
+        auto arg_src = CTX_IN_SYCL_MEMORY(DNNL_ARG_SRC);
+        auto arg_weights = CTX_IN_SYCL_MEMORY(DNNL_ARG_WEIGHTS);
+        auto arg_dst = CTX_OUT_SYCL_MEMORY(DNNL_ARG_DST);
+
+        sycl_memory_arg<::sycl::access::mode::read> arg_bias;
+        sycl::sycl_memory_storage_base_t *temp_dst_mem;
+        sycl::sycl_memory_storage_base_t *temp_reorder_mem;
+
         std::shared_ptr<scratch_acc_t> scratch_acc;
         std::shared_ptr<scratch_acc_t> filter_scratch_acc;
-        std::shared_ptr<scratch_acc_t> temp_dst_acc;
-        std::shared_ptr<scratch_acc_t> temp_reorder_acc;
 
         const bool use_temp_dst = pd()->use_temp_dst();
 
         if (with_scratchpad) {
-            scratch_acc = std::make_shared<scratch_acc_t>(
-                    utils::downcast<sycl::sycl_buffer_memory_storage_t *>(
-                            ctx.get_scratchpad_grantor()
-                                    .get_memory_storage(memory_tracking::names::
-                                                    key_conv_cudnn_algo)
-                                    .get())
-                            ->buffer()
-                            .get_access<::sycl::access::mode::read_write>(cgh));
-        }
-        if (with_bias) {
-            bias_acc = std::make_shared<
-                    ::sycl::accessor<uint8_t, 1, ::sycl::access::mode::read>>(
-                    CTX_IN_ACCESSOR(DNNL_ARG_BIAS));
+            scratch_acc = std::make_shared<scratch_acc_t>(CTX_SCRATCH_ACCESSOR(
+                    memory_tracking::names::key_conv_cudnn_algo));
         }
+        if (with_bias) { arg_bias = CTX_IN_SYCL_MEMORY(DNNL_ARG_BIAS); }
         if (pd()->impl_->using_transformed_filter()) {
             filter_scratch_acc
                     = std::make_shared<scratch_acc_t>(CTX_SCRATCH_ACCESSOR(
                             memory_tracking::names::key_conv_cudnn_filter));
         }
 
+        sycl_memory_arg<::sycl::access::mode::read_write> temp_dst;
+        sycl_memory_arg<::sycl::access::mode::read_write> temp_reorder;
+
         if (use_temp_dst) {
-            temp_dst_acc = std::make_shared<scratch_acc_t>(
-                    buffer(scratch_storage.get())
-                            .get_access<::sycl::access::mode::read_write>(cgh));
-            temp_reorder_acc = std::make_shared<scratch_acc_t>(
-                    buffer(scratch_storage_2.get())
-                            .get_access<::sycl::access::mode::read_write>(cgh));
+            temp_dst_mem = utils::downcast<sycl::sycl_memory_storage_base_t *>(
```

Comment:
`sycl_memory_arg_t` now takes `memory_storage_t` so you can pass `scratch_storage.get()` directly without casting to `sycl_memory_storage_base_t`.